modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2591,7 +2591,12 @@ Error GDScriptCompiler::_populate_class_members(GDScript *p_script, const GDScri
 					_set_error(vformat(R"(Could not find class "%s" in "%s".)", base->fully_qualified_name, base->path), nullptr);
 					return ERR_COMPILATION_FAILED;
 				}
-				ERR_FAIL_COND_V(!base->is_valid() && !base->reloading, ERR_BUG);
```

Comment:
This is what I'm worried about! I haven't noticed the absence of this causing problems in the small tests I have done, but I'd love to see this merged so we can fix any regressions this might lead to.

Reply:
I believe that the checks above should cause the function to return early in the case of errors so that this isn't necessary. But you never know when things are this subtle :)